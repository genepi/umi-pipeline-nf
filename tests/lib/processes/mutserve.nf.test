nextflow_process {

    name "Test Process MUTSERVE"
    script "lib/processes/mutserve.nf"
    process "MUTSERVE"

    test("Should run without failures") {

        when {
            params {
                variant_calling = true 
                variant_caller = "mutserve"
            }
            process {
                """
                input[0] = Channel.of(["barcode03", "final", "$baseDir/tests/input/variant_calling/barcode03/final.bam", "$baseDir/tests/input/variant_calling/barcode03/final.bam.bai"])
                input[1] = Channel.of("final")
                input[2] = file("$baseDir/tests/input/pipeline/ref/lpa-ref2645.bed")
                input[3] = file("$baseDir/tests/input/pipeline/ref/lpa-ref2645.fasta")
                input[4] = file("$baseDir/tests/input/pipeline/ref/lpa-ref2645.fasta.fai")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }
    }

    test("Should output three files") {

        when {
            params {
                variant_calling = true 
                variant_caller = "mutserve"
            }
            process {
                """
                input[0] = Channel.of(
                    ["barcode03", "final", "$baseDir/tests/input/variant_calling/barcode03/final.bam", "$baseDir/tests/input/variant_calling/barcode03/final.bam.bai"],
                    ["barcode04", "final", "$baseDir/tests/input/variant_calling/barcode04/final.bam", "$baseDir/tests/input/variant_calling/barcode04/final.bam.bai"],
                    ["barcode05", "final", "$baseDir/tests/input/variant_calling/barcode05/final.bam", "$baseDir/tests/input/variant_calling/barcode05/final.bam.bai"]
                    )
                input[1] = Channel.of("final", "final", "final")
                input[2] = file("$baseDir/tests/input/pipeline/ref/lpa-ref2645.bed")
                input[3] = file("$baseDir/tests/input/pipeline/ref/lpa-ref2645.fasta")
                input[4] = file("$baseDir/tests/input/pipeline/ref/lpa-ref2645.fasta.fai")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }
}
