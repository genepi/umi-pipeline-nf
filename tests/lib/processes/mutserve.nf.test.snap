{
    "Should output three files": {
        "content": [
            {
                "0": [
                    "final.txt:md5,1e5554e27b3b8a0687374df975a6c9cb",
                    "final.txt:md5,871ecfddc44cb8b3aee5a3c482fa2900",
                    "final.txt:md5,bdcd9fa23bcd1e226b07988460f9b162"
                ],
                "1": [
                    "final_raw.txt:md5,94352f26d9095f38eb44ff730db75dbc",
                    "final_raw.txt:md5,a647c13b88e318322308fa70526a8d96",
                    "final_raw.txt:md5,f8da170e2122c7caffa8fb3efb311434"
                ],
                "variants": [
                    "final.txt:md5,1e5554e27b3b8a0687374df975a6c9cb",
                    "final.txt:md5,871ecfddc44cb8b3aee5a3c482fa2900",
                    "final.txt:md5,bdcd9fa23bcd1e226b07988460f9b162"
                ],
                "variants_raw": [
                    "final_raw.txt:md5,94352f26d9095f38eb44ff730db75dbc",
                    "final_raw.txt:md5,a647c13b88e318322308fa70526a8d96",
                    "final_raw.txt:md5,f8da170e2122c7caffa8fb3efb311434"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.0"
        },
        "timestamp": "2024-03-11T12:49:45.12999"
    },
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "final.txt:md5,1e5554e27b3b8a0687374df975a6c9cb"
                ],
                "1": [
                    "final_raw.txt:md5,94352f26d9095f38eb44ff730db75dbc"
                ],
                "variants": [
                    "final.txt:md5,1e5554e27b3b8a0687374df975a6c9cb"
                ],
                "variants_raw": [
                    "final_raw.txt:md5,94352f26d9095f38eb44ff730db75dbc"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.0"
        },
        "timestamp": "2024-03-11T12:49:34.341905"
    }
}